gettsim/visualization.py

```diff
@@ -571,3 +331,246 @@ def _kth_order_successors(dag, node, order):
+            ayref="y",
+            ax=df["x0"][i],
+            ay=df["y0"][i],
+            arrowhead=2,
+            arrowsize=2,
+            startstandoff=5,
+            standoff=5,
+            arrowcolor="gray",
+        )
+        arrows.append(arrow)
+
+    # plot the nodes, edges and arrows together
+
+    fig = go.FigureWidget(
+        layout=go.Layout(
+            showlegend=True,
+            hovermode="closest",
+            clickmode="select",
+            annotations=arrows,
+            hoverlabel_font_size=10,
+            margin={"b": 20, "l": 5, "r": 5, "t": 40},
+            xaxis={"showgrid": False, "zeroline": False, "showticklabels": False},
+            yaxis={"showgrid": False, "zeroline": False, "showticklabels": False},
+        )
+    )
+    fig.add_scatter(
+        x=edge_x,
+        y=edge_y,
+        line={"width": 0.5, "color": "blue"},
+        hoverinfo="none",
+        mode="lines",
+        showlegend=False,
+    )
+    # choose the different option for plotting
+
+    if (len(names) > 10) or (showlabels is False):
+        mode = "markers"
+        fig.add_scatter(
+            x=combo.x,
+            y=combo.y,
+            mode=mode,
+            hoverinfo="text",
+            textposition="bottom center",
+            text=list(names),
+            showlegend=showlegend,
+            marker={
+                "showscale": False,
+                "reversescale": True,
+                "color": "red",
+                "size": 15,
+            },
+        )
+
+    elif (len(names) <= 10) or ((len(names) > 10) and (showlabels is True)):
+        mode = "markers+text"
+
+    elif (len(names) > 10) and (showlabels is True):
+        mode = "markers +text"
+
+    if source_code is True:
+        for i in range(len(combo)):
+            fig.add_scatter(
+                x=[combo.x[i]],
+                y=[combo.y[i]],
+                mode=mode,
+                hovertext=combo.source_code[i],
+                hoverinfo="text",
+                textposition="bottom center",
+                hoverlabel={"bgcolor": "lightgrey", "font": {"color": "black"}},
+                text=names[i],
+                showlegend=False,
+                name=names[i],
+                marker={
+                    "showscale": False,
+                    "reversescale": True,
+                    "color": "red",
+                    "size": 15,
+                },
+            )
+
+    elif source_code is False:
+        fig.add_scatter(
+            x=combo.x,
+            y=combo.y,
+            mode=mode,
+            hoverinfo="skip",
+            textposition="bottom center",
+            text=list(names),
+            showlegend=False,
+            marker={
+                "showscale": False,
+                "reversescale": True,
+                "color": "red",
+                "size": 15,
+            },
+        )
+    else:
+
+        raise ValueError("Please choose right argument specification")
```

Comment:
Be more specific about which argument is wrong, which argument value was given, and which argument values are allowed.